### PR TITLE
Add unit testing

### PR DESCRIPTION
**Pull request by schmidtwmark**
Adds some explicit testing for discriminants, empty enum, and other cases

### CODE REVIEW

**Comment by j-vanderstoep** [comment]
Friendly ping! Any concerns with adding these unit tests?

**Review by dtolnay — CHANGES_REQUESTED** [review]
- Please delete `cfg(test)`. `tests/` are only ever compiled in test mode.
- Please delete `macro_use`. There are no macros in this module, or anything to depend on them.
- Please delete `mod tests`.
- Please format the code with rustfmt.
- Please delete all the derived `Eq` impls, they are not used anywhere.

**Comment by dtolnay** [comment]
@j-vanderstoep it would be more useful for you to provide a first-pass review rather than pinging. There are a number of issues that I think should have been apparent to you.

**Comment by schmidtwmark** [comment]
Addressed feedback.

File: tests/test.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+
```
Comment (dtolnay) [inline tests/test.rs]:
```suggestion
```

**Review by dtolnay — CHANGES_REQUESTED** [review]

**Review by dtolnay — APPROVED** [review]
Thanks!

**Comment by j-vanderstoep** [comment]
> @j-vanderstoep it would be more useful for you to provide a first-pass review rather than pinging. There are a number of issues that I think should have been apparent to you.

Agreed. Really sorry about that. I incorrectly put this into my "I'm blocked on review" queue.

**Pull request merged**